リスト部分の追加

freemake-061 opened the pull request:
## チケットへのリンク
https://github.com/freemake-061/AppTask/commit/2c2b71db5cb52e7996f745c703377216291853af
## 概要
アプリ製作課題1
## 完了条件
- ConstantsクラスがConstants.kt内にあること
- Stockがリストに表示されること
- リストの行を奇数偶数で色分けできること
## 実現方法
- MainActivity.kt内のConstantsクラスを、新規作成したConstants.kt内に移す
- StockData.ktにテスト用のStockを用意し、StockListでStockCardを並べる
- indexをStockCardに渡し、奇数偶数を判定して色を変える
## 想定される問題点
なし
## その他
なし

freemake-061 committed:
定数をConstants.ktへ移動
ストックをリストで管理&表示
リストを奇数偶数で色分け

EitoSasaki commented on app/src/main/java/com/example/apptask/StockData.kt:
```diff
@@ -0,0 +1,12 @@
+
+import androidx.compose.runtime.mutableStateListOf
+
+object StockData {
```
この`object `は多分不要になりますので削除してください。
`Stock`のリストをMainActivityに定義して、`StockList`まで値渡しするようにしてください。

EitoSasaki commented on app/src/main/java/com/example/apptask/StockData.kt:
```diff
@@ -0,0 +1,12 @@
+import androidx.compose.runtime.mutableStateListOf
+
+object StockData {
+    var stocks = mutableStateListOf(
```
`State`はComposable内で値の変化を検知したい場合に使用するので、
Composable内に定義しないなら、`State`にする必要はないです。

EitoSasaki commented on app/src/main/java/com/example/apptask/StockList.kt:
```diff
@@ -19,35 +22,54 @@ import androidx.compose.runtime.setValue
 
 @Composable
-fun StockList() {
+fun StockCard(ind: Int, stc: Stock) {
```
省略形での命名は、一般的なパターン以外では使用しないほうがいいです。
冗長だと思っても、誰がみても意図が伝わることを優先してください。

freemake-061 committed:
修正
StockData.ktからMainActivity.ktへstocks変数を移動
mutableStateListOfからmutableListOfへ変更
ind→index、stc→stockへ命名を変更
